### ISSUE TITLE
Chat messages private/public will dissapear if a user has been promoted/demoted and a private chat is opened with the user

### ISSUE THREAD

**Issue by Subdarub** [bigbluebutton/bigbluebutton#12738]
**Describe the bug**
If a user has been promoted or demoted (Moderator <-> Participant), and someone creates a private chat with said user. Every message that was previously present for the user be it private or public will dissapear. 

**To Reproduce**
1. Create a session with multiple users
2. Type multiple messages in chat
3. Promote one of the users to moderator
4. Create a private chat with the user

**Expected behavior**
The messages are still present

**Actual behavior**
The messages are gone

**BBB version:**
2.3.4  (Build 1783)
demo.bigbluebutton.org  (Build 1795)

**Desktop (please complete the following information):**
 - Windows 10 21H1
 - Chromium Edge 91.0.864.64

**Additional Context**
If you are unable to reproduce this behavior, i can create a video of it happening.

**Comment by ramonlsouza** [comment]
Duplicated (#13192)

**Comment by antobinary** [comment]
Hi @Subdarub 
Thank you for reporting this! Somehow I missed your report and we recently encountered the issue independently, including a fix from @Tainan404 https://github.com/bigbluebutton/bigbluebutton/pull/13193
This fix is going to 2.4 first and will (upon some further testing) be backported to 2.3.

Thanks again for reporting this - it was a tricky one to catch in our case and had I spotted your report ealier, we would have had easier time. We appreciate your contributions! :trophy:
